src/flyte/_input_context.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import contextvars
+from contextlib import asynccontextmanager, contextmanager
+from typing import AsyncIterator, Dict, Iterator
+
+from flyte._context import ctx
+
+# Context variable to store the input context
+# This stores both global context (from with_runcontext) and local context (from context managers)
+_input_context_var: contextvars.ContextVar[Dict[str, str]] = contextvars.ContextVar("input_context", default={})
```

Comment:
Why do we need this? 
Input context should only be on the task right? Global context is dangerous and will not work for actors. 
Let's just use taskcontext.

To pass it down, you can modify the taskcontext data. This is what i was going to write